src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs

```diff
@@ -110,46 +110,60 @@ public override bool RunTask ()
+					sw.WriteLine ($"--enummetadata=\"{EnumMetadataDirectory}\"");
+				if (!string.IsNullOrEmpty (AssemblyName))
+					sw.WriteLine ($"--assembly={AssemblyName}");
+
+				if (!NoStdlib) {
+					string fxpath = MonoAndroidFrameworkDirectories.Split (';').First (p => new DirectoryInfo (p).GetFiles ("mscorlib.dll").Any ());
+					sw.WriteLine ($"--ref=\"{Path.Combine (Path.GetFullPath (fxpath), "mscorlib.dll")}\"");
+				}
 
-			cmd.AppendSwitchIfNotNull ("--codegen-target=", CodegenTarget);
-			cmd.AppendSwitchIfNotNull ("--csdir=", OutputDirectory);
-			cmd.AppendSwitchIfNotNull ("--enumdir=", EnumDirectory);
-			cmd.AppendSwitchIfNotNull ("--enummetadata=", EnumMetadataDirectory);
-			cmd.AppendSwitchIfNotNull ("--assembly=", AssemblyName);
+				if (ReferencedManagedLibraries != null)
+					foreach (var lib in ReferencedManagedLibraries)
+						sw.WriteLine ($"--ref=\"{Path.GetFullPath (lib.ItemSpec)}\"");
+				if (AnnotationsZipFiles != null)
+					foreach (var zip in AnnotationsZipFiles)
+						sw.WriteLine ($"--annotations=\"{Path.GetFullPath (zip.ItemSpec)}\"");
 
-			if (!NoStdlib) {
-				string fxpath = MonoAndroidFrameworkDirectories.Split (';').First (p => new DirectoryInfo (p).GetFiles ("mscorlib.dll").Any ());
-				cmd.AppendSwitchIfNotNull ("--ref=", Path.Combine (Path.GetFullPath (fxpath), "mscorlib.dll"));
-			}
-			
-			if (ReferencedManagedLibraries != null)
-				foreach (var lib in ReferencedManagedLibraries)
-					cmd.AppendSwitchIfNotNull ("--ref=", Path.GetFullPath (lib.ItemSpec));
-			if (AnnotationsZipFiles != null)
-				foreach (var zip in AnnotationsZipFiles)
-					cmd.AppendSwitchIfNotNull ("--annotations=", Path.GetFullPath (zip.ItemSpec));
+				foreach (var tf in transform_files)
+					sw.WriteLine ($"--{tf.Item2}={tf.Item1}");
 
-			foreach (var tf in transform_files)
-				cmd.AppendSwitchIfNotNull (string.Format ("--{0}=", tf.Item2), tf.Item1);
+				if (!string.IsNullOrEmpty (AndroidApiLevel))
+					sw.WriteLine ($"--api-level={AndroidApiLevel}");
 
-			cmd.AppendSwitchIfNotNull ("--api-level=", AndroidApiLevel);
+				if (!string.IsNullOrEmpty (TypeMappingReportFile))
+					sw.WriteLine ($"--type-map-report=\"{TypeMappingReportFile}\"");
 
-			cmd.AppendSwitchIfNotNull ("--type-map-report=", TypeMappingReportFile);
+				sw.WriteLine ("--global");
+				sw.WriteLine ("--public");
 
-			cmd.AppendSwitch ("--global");
-			cmd.AppendSwitch ("--public");
+				if (UseShortFileNames)
+					sw.WriteLine ("--use-short-file-names");
 
-			if (UseShortFileNames)
-				cmd.AppendSwitch ("--use-short-file-names");
-
-			if (EnableInterfaceMembersPreview && SupportsCSharp8)
-				cmd.AppendSwitch ("--lang-features=interface-constants,default-interface-methods");
+				if (EnableInterfaceMembersPreview && SupportsCSharp8)
+					sw.WriteLine ("--lang-features=interface-constants,default-interface-methods");
+			}
 
+			Log.LogDebugMessage ("[Generator] response file {0}: {1}", responseFile, File.ReadAllText (responseFile));
```

Comment:
Instead of reading the file back into memory, could we make a helper method that logs and writes for each call?

Something like:
```csharp
LogDebugMessage ("[Generator] response file: {0}", responseFile);

WriteLine (sw, "--global");
WriteLine (sw, "--public");

// The other switches...

void WriteLine (StreamWriter sw, string line)
{
    sw.WriteLine (line);
    Log.LogDebugMessage (line);
}
```

Reply:
Yup I thought about doing that lol.